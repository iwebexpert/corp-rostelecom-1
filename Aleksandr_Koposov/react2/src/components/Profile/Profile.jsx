import React, { useEffect, useState } from 'react'

import {
  Card,
  CardContent,
  CardActions,
  Badge,
  Icon,
  Avatar,
  TextField,
  Button,
  Dialog,
  DialogTitle,
  CircularProgress
} from '@material-ui/core'

import './Profile.scss'

export const Profile = (props) => {
  const { user, isLoading } = props
  const { id, name, age, about, avatar } = user || {}
  const [profileState, setProfile] = useState(
    {
      init: false,
      dialog: false,
      id,
      name,
      age,
      about,
      avatar
    }
  )

  useEffect(() => {
    setProfile({
      ...profileState,
      init: true,
      ...props.user
    })
  }, [])

  const onInputChange = (event) => {
    setProfile({
      ...profileState,
      [event.target.name]: event.target.value
    })
  }

  const saveChanges = () => {
    const { onSave } = props
    const { id, name, age, about, avatar } = profileState
    onSave({ id, name, age, about, avatar })
  }

  const toggleDialog = () => {
    setProfile({
      ...profileState,
      dialog: !profileState.dialog
    })
  }

  const avatarSize = {
    width: 128,
    height: 128
  }

  return (
    <Card className="profile" elevation={3}>
      <CardContent>
        <Badge
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          badgeContent={<Icon>edit</Icon>}
          onClick={toggleDialog}
        >
          {!isLoading ? (
            <Avatar
              style={avatarSize}
              alt="Аватар пользователя"
              src={profileState.avatar}
            />
          ) : (
              <Avatar style={avatarSize}>
                <CircularProgress />
              </Avatar>
            )}
        </Badge>
        <Dialog
          onClose={toggleDialog}
          open={profileState.dialog}
          className="avatar-dialog"
        >
          <DialogTitle>Выбор аватара</DialogTitle>
          <Card>
            <CardContent>
              <TextField
                name="avatar"
                label="Ссылка на аватар (можете изменять число в конце для теста)"
                variant="outlined"
                value={profileState.avatar}
                onChange={onInputChange}
                style={{ width: '550px' }}
              />
            </CardContent>
          </Card>
        </Dialog>
        {!isLoading ?
          (
            <div>
              <TextField
                name="name"
                label="Имя"
                variant="outlined"
                value={profileState.name}
                onChange={onInputChange}
              />
              <TextField
                name="age"
                label="Возраст"
                variant="outlined"
                type="number"
                value={profileState.age}
                onChange={onInputChange}
              />
              <TextField
                name="about"
                label="О себе"
                multiline
                variant="outlined"
                value={profileState.about}
                onChange={onInputChange}
              />
            </div>
          ) : (
            <div style={{
              width: '100%',
              textAlign: 'center'
            }}>
              <CircularProgress />
            </div>
          )}
      </CardContent>
      <CardActions>
        <Button
          color="primary"
          onClick={saveChanges}
        >Сохранить</Button>
      </CardActions>
    </Card>
  )
}
