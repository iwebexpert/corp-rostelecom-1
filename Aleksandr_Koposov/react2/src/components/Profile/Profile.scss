.profile {
  & .MuiBadge-badge {
    height: 36px;
    width: 36px;
    border-radius: 50%;
    background-color: grey;
    color: white;
    cursor: pointer;
  }

  & .MuiTextField-root {
    width: 100%;
    margin: 15px 0;
  }
}

.avatar-dialog {
  & .MuiTextField-root {
    max-width: 550px;
    width: 100%;
  }
}
