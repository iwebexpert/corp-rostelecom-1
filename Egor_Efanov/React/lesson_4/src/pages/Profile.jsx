import React from 'react';

export function Profile() {
    return (
        <div><p>Profile</p>
            <ul>
                <li>
                    Name: WebDev
                </li>
                <li>
                    Age: 25
                </li>
                <li>
                    From: Samara
                </li>

            </ul>
        </div>
    );
} 