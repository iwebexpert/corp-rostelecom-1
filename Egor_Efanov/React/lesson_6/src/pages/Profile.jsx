import React from 'react';
import { profile } from 'helpers/profileData';

export function Profile() {
    return (
        <div>Profile page
            <ul>
                <li>
                    Name: {profile[0].name}
                </li>
                <li>
                    Age: {profile[0].age}
                </li>
                <li>
                    From: {profile[0].from}
                </li>

            </ul>
        </div>
    );
} 