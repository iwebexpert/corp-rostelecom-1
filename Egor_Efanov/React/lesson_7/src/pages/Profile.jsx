import React, { Component } from 'react';
import { Button, TextField } from '@material-ui/core';


export class Profile extends Component {
    render() {
        const profile = this.props.profile;
        console.log(profile);
        const { isLoading, isError } = this.props;

        if (isError) {
            return (<div>Не удалось загрузить профиль...</div>);
        }

        if (isLoading) {
            return (<div>Loading...</div>);
        }
        return (
            <div>Profile page
                <ul>
                    <li>
                        Name: {profile[0].name}
                    </li>
                    <li>
                        Age: {profile[0].age}
                    </li>
                    <li>
                        From: {profile[0].from}
                    </li>

                </ul>
            </div>
        );
    }

} 