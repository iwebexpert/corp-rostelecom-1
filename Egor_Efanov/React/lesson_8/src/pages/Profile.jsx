import React, { Component } from 'react';
import { Button, TextField } from '@material-ui/core';


export class Profile extends Component {
    render() {
        const profile = this.props.profile;
        console.log(profile);
        const { isLoading, isError } = this.props;

        if (isError) {
            return (<div>Не удалось загрузить профиль...</div>);
        }

        if (isLoading) {
            return (<div>Loading...</div>);
        }
        return (
            <div>Profile page
                <ul>
                    <li>
                        Name: {profile.name}
                    </li>
                    <li>
                        Age: {profile.age}
                    </li>
                    <li>
                        From: {profile.from}
                    </li>

                </ul>
            </div>
        );
    }

} 