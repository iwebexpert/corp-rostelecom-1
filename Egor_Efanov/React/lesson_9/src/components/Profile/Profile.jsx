import React from 'react';

export const Profile = ({ name, age, email, isError, isLoading }) => {

    if (isError) {
        return (<div>Не удалось загрузить профиль пользователя...</div>);
    }

    if (isLoading) {
        return (<div>Loading...</div>);
    }

    return (<div >
        <ul >
            <li>Имя: {name}</li>
            <li>Возраст: {age}</li>
            <li>from: {email}</li>
        </ul>
    </div>);
}; 