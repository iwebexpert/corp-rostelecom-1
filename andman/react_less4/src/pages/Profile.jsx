import React from 'react';

export function ProfilePage() {
    return (
        <div>
            <h2>Profile page</h2>
            E-mail:  test@mail.ru <hr />
            Логин: Test1 <hr />
        </div>


    );
}