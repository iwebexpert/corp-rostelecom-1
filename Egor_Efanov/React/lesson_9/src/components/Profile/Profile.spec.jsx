import React from 'react';
import { Profile } from './Profile';

describe('Profile component', () => {
    it('should render Profile component (1)', () => {
        const component = shallow(<Profile name='WebDev' age='35' from='Moscow' />);
        expect(component).toMatchSnapshot();
    });

    it('should render MessageList component (2)', () => {
        const component = render(<Profile name='WebDev' age='35' from='Moscow' />);
        expect(component).toMatchSnapshot();
    });
}); 