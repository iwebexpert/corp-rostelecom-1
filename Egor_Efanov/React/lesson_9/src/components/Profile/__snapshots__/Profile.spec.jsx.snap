// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render MessageList component (2) 1`] = `
<div>
  <ul>
    <li>
      Имя: WebDev
    </li>
    <li>
      Возраст: 35
    </li>
    <li>
      from: 
    </li>
  </ul>
</div>
`;

exports[`Profile component should render Profile component (1) 1`] = `
<div>
  <ul>
    <li>
      Имя: 
      WebDev
    </li>
    <li>
      Возраст: 
      35
    </li>
    <li>
      from: 
    </li>
  </ul>
</div>
`;
